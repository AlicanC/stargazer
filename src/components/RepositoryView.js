// @flow

import * as React from 'react';
import { graphql, compose } from 'react-apollo';
import gql from 'graphql-tag';
import { Row, Col } from 'reactstrap';

type Props = {
  repository: Object,
  addStar: Function,
  removeStar: Function,
};

function RepositoryView(props: Props) {
  const { repository, addStar, removeStar } = props;
  const { name, url, viewerHasStarred } = repository;

  return (
    <Row>
      <Col>
        <a href={url}>{name}</a>
      </Col>
      <Col style={{ textAlign: 'right' }}>
        {viewerHasStarred ? (
          <button onClick={removeStar}>Starred</button>
        ) : (
          <button onClick={addStar}>Not Starred</button>
        )}
      </Col>
    </Row>
  );
}

RepositoryView.fragments = {
  repository: gql`
    fragment RepositoryViewRepository on Repository {
      id
      name
      url
      viewerHasStarred
    }
  `,
};

export default compose(
  graphql(
    gql`
      mutation AddStarMutation($input: AddStarInput!) {
        addStar(input: $input) {
          starrable {
            id
            viewerHasStarred
          }
        }
      }
    `,
    {
      name: 'addStar',
      options: (props) => ({
        variables: {
          input: {
            starrableId: props.repository.id,
          },
        },
      }),
    },
  ),
  graphql(
    gql`
      mutation RemoveStarMutation($input: RemoveStarInput!) {
        removeStar(input: $input) {
          starrable {
            id
            viewerHasStarred
          }
        }
      }
    `,
    {
      name: 'removeStar',
      options: (props) => ({
        variables: {
          input: {
            starrableId: props.repository.id,
          },
        },
      }),
    },
  ),
)(RepositoryView);
