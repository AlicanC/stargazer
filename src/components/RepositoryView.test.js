import React from 'react';
import TestRenderer from 'react-test-renderer';
import { MockedProvider } from 'react-apollo/test-utils';

import RepositoryView from './RepositoryView';

// jest.mock('react-router-dom', () => ({
//   Link: 'div',
// }));

// jest.mock('react-redux', () => ({
//   connect: () => (C) => C,
// }));

it('matches snapshot', () => {
  const repository = {
    id: 'MDEwOlJlcG9zaXRvcnkxMDI3MDI1MA==',
    nameWithOwner: 'facebook/react',
    url: 'https://github.com/facebook/react',
    viewerHasStarred: true,
    description:
      'A declarative, efficient, and flexible JavaScript library for building user interfaces.',
    hasIssuesEnabled: true,
    issues: { totalCount: 6373, __typename: 'IssueConnection' },
    pullRequests: { totalCount: 7262, __typename: 'PullRequestConnection' },
    stargazers: { totalCount: 112426, __typename: 'StargazerConnection' },
    __typename: 'Repository',
  };

  const renderer = TestRenderer.create(
    <MockedProvider>
      <RepositoryView repository={repository} />
    </MockedProvider>,
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});
