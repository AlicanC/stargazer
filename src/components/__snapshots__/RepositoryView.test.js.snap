// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="mb-3 row"
  >
    <div
      className="col-7 col-md-9"
    >
      <div
        className="row"
      >
        <div
          className="col-md-auto"
        >
          <a
            className="css-542wex ec628oh0"
            href="https://github.com/facebook/react"
          >
            facebook/react
          </a>
        </div>
        <div
          className="col-12 col-md-5"
        >
          <span>
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-star fa-w-18 "
              data-icon="star"
              data-prefix="fas"
              role="img"
              style={Object {}}
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             
            112k
          </span>
          <span
            className="ml-2"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-exclamation-triangle fa-w-18 "
              data-icon="exclamation-triangle"
              data-prefix="fas"
              role="img"
              style={Object {}}
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             
            6k
          </span>
          <span
            className="ml-2"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-code-branch fa-w-12 "
              data-icon="code-branch"
              data-prefix="fas"
              role="img"
              style={Object {}}
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M384 144c0-44.2-35.8-80-80-80s-80 35.8-80 80c0 36.4 24.3 67.1 57.5 76.8-.6 16.1-4.2 28.5-11 36.9-15.4 19.2-49.3 22.4-85.2 25.7-28.2 2.6-57.4 5.4-81.3 16.9v-144c32.5-10.2 56-40.5 56-76.3 0-44.2-35.8-80-80-80S0 35.8 0 80c0 35.8 23.5 66.1 56 76.3v199.3C23.5 365.9 0 396.2 0 432c0 44.2 35.8 80 80 80s80-35.8 80-80c0-34-21.2-63.1-51.2-74.6 3.1-5.2 7.8-9.8 14.9-13.4 16.2-8.2 40.4-10.4 66.1-12.8 42.2-3.9 90-8.4 118.2-43.4 14-17.4 21.1-39.8 21.6-67.9 31.6-10.8 54.4-40.7 54.4-75.9zM80 64c8.8 0 16 7.2 16 16s-7.2 16-16 16-16-7.2-16-16 7.2-16 16-16zm0 384c-8.8 0-16-7.2-16-16s7.2-16 16-16 16 7.2 16 16-7.2 16-16 16zm224-320c8.8 0 16 7.2 16 16s-7.2 16-16 16-16-7.2-16-16 7.2-16 16-16z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             
            7k
          </span>
        </div>
      </div>
    </div>
    <div
      className="col-5 col-md-3"
      style={
        Object {
          "textAlign": "right",
        }
      }
    >
      <button
        className="btn btn-warning"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star fa-w-18 "
          data-icon="star"
          data-prefix="fas"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         Starred
      </button>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col"
    >
      <p>
        A declarative, efficient, and flexible JavaScript library for building user interfaces.
      </p>
    </div>
  </div>,
]
`;
